packages/dashboard/src/components/resizable-panes/resizable-panes.tsx

```diff
@@ -0,0 +1,108 @@
+import { Component, State, Listen, h } from '@stencil/core';
+
+const HANDLE_WIDTH_PX = 10;
+const LEFT_WIDTH_PERCENT = 0.15;
+const CENTER_WIDTH_PERCENT = 0.7;
+const RIGHT_WIDTH_PERCENT = 0.15;
+
+@Component({
+  tag: 'resizable-panes',
+  styleUrl: 'resizable-panes.css',
+})
+export class ResizablePanes {
+  /** Currently active drag hangle during a drag, or null if not dragging */
+  @State() currentDragHandle: 'left' | 'right' | null = null;
+
+  /** Last seen mouse x position during a drag, in px from screen left side */
+  @State() lastSeenAtX: number | null = null;
+
+  /** Total x distance moved during a drag, in px */
+  @State() movedX: number | null = null;
+
+  /** Current widths of the three panes, in percentage of screen width */
+  @State() leftPaneWidth: number = window.innerWidth * LEFT_WIDTH_PERCENT;
+  @State() centerPaneWidth: number = window.innerWidth * CENTER_WIDTH_PERCENT;
+  @State() rightPaneWidth: number = window.innerWidth * RIGHT_WIDTH_PERCENT;
```

Comment:
will this still work if the dashboard isn't utilized as a full page widget?

Reply:
No it won't - great point. I pushed a revised version that determines these by measuring the parent element instead. 

As a bonus, in the process I replaced a bunch of JS with CSS grid, so the component is simpler now.

There's only one downside: to measure the actual width of the parent element, the component needs to render, measure, and then re-render. Stencil complains about this with console warnings about unnecessary re-renders.